symengine/logic.cpp

```diff
@@ -478,6 +478,202 @@ RCP<const Boolean> logical_xor(const vec_boolean &s)
     }
 }
 
+Relational::Relational(const RCP<const Basic> &lhs, const RCP<const Basic> &rhs)
+    : TwoArgBasic<Boolean>(lhs, rhs)
+{
+}
+
+hash_t Relational::__hash__() const
+{
+    hash_t seed = this->get_type_code();
+    hash_combine<Basic>(seed, *get_arg1());
+    hash_combine<Basic>(seed, *get_arg2());
+    return seed;
+}
+
+inline bool Relational::__eq__(const Basic &o) const
```

Comment:
This has to be overloaded, as the case `eq(Eq(x, y), Eq(y, x))` is not considered in `TwoArgBasic`.